### DISCUSSION TITLE
Блокировка таблиц при миграции

### DISCUSSION THREAD

**Discussion by SugarFreeOrbit** [totumonline/totum-ru-issues-and-discussions#240] [Как это сделать?] [answered]
Дана некая таблица. В этой таблице уже есть определённое количество строк (потенциально довольно большое). Пользователь системы через Totum пытается добавить к этой таблице новый столбец. Как я понимаю под капотом запускается миграция. В это время запускается старый Код работающий с этой таблицей, например, он пытается сделать INSERT. Как себя поведёт система в этом случае? Код будет "ждать" разблокировки таблицы или произойдёт ошибка?

**Comment by ttm-alexeypolunin** [comment, marked as answer]
Простой пользователь не может добавить Столбец. Только Создатель или если вы написали код-действия доступный простому пользователю (что является очень плохой практикой).

Никакой миграции при добавлении столбца нет — просто добавляется столбец в БД (для Простых и Циклов) по транзакции и данные об этом столбце в таблицу Состав таблиц (для временных и расчетных только это).

Если в момент добавления столбца происходит insert в таблицу в которую столбец добавляется, то операция дождется снятия лока с таблицы и запишет строку игнорируя новый столбец.

Если в момент добавляения столбца произошло изменение апдейт-кода таблицы, то транзакция с insert будет перезапущена в соответсвии с настройками [Актуальности](https://ru.docs.totum.online/tables#table-settings-actual) таблицы.
